include/cuco/detail/open_addressing/kernels.cuh

```diff
@@ -354,13 +362,21 @@ CUCO_KERNEL void count(InputIt first, cuco::detail::index_type n, AtomicT* count
   auto idx               = cuco::detail::global_thread_id() / CGSize;
 
   while (idx < n) {
-    auto const key = *(first + idx);
+    typename std::iterator_traits<InputIt>::value_type const& key = *(first + idx);
```

Comment:
caveats: `auto const` should **_not_** be used in any kernels when dereferencing input since it cannot deal with thrust wrappers properly.